src/solvers/refinement/string_constraint_generator_insert.cpp

```diff
@@ -150,6 +152,7 @@ exprt string_constraint_generatort::add_axioms_for_insert_int(
 /// \param f: function application with three arguments: a string, an
 ///   integer offset, and a Boolean
 /// \return a new string expression
+DEPRECATED("should convert the value to string and call insert")
```

Comment:
Am I right when saying that some bits are marked deprecated here when really they just need to get onto someone's task list to fix?

Reply:
I mainly wrote this to make it easier to fix code that uses deprecated functionality as we go and to prevent people from writing new code with deprecated functionality, but if someone took the time to fix all of the instances where those are used that would of course also be helpful.

Reply:
This is correctly marked deprecated but the message is not totally clear: instead of calling this function we can obtain the same result by using calls to `add_axioms_from_boolean` and `add_axioms_for_insert`.